azure-pipelines.yml

```diff
@@ -274,19 +274,19 @@ stages:
             Download-OpenBLAS('1')
         }
       displayName: 'Download / Install OpenBLAS'
-    - script: |
-        choco install -y llvm
-        set PATH=C:\Program Files\LLVM\bin;%PATH%
-        echo '##vso[task.setvariable variable=PATH]%PATH%'
-      displayName: 'Install clang-cl'
-    - script: |
-        clang-cl.exe --version
-      displayName: 'clang-cl version'
+
     - powershell: |
         # wheels must use same version
-        choco install -y mingw --forcex86 --force --version=7.3.0
+        choco install -y mingw rtools
+      displayName: 'Install 64-bit mingw for 64-bit builds'
+      condition: and(succeeded(), eq(variables['BITS'], 64))
+
+    - powershell: |
+        # wheels must use same version
+        choco install -y mingw --forcex86 --force --version=8.1.0
```

Comment:
I don't think that the setup.py build actually used clang-cl as the c-compiler. If I look at an azure build log, e.g.

https://dev.azure.com/scipy-org/SciPy/_build/results?buildId=23257&view=logs&j=203f29d8-a99b-512b-bad4-9c985ff1792a&t=549b98f4-8390-5ed6-827a-12915e592e34&l=9317

the existing builds appear to use MSVC for the compilation. Or is that an incorrect take?

Reply:
I think that choco installing mingw probably might not required for the 64-bit build, the (gfortran, gcc) compilers are actually installed via mingw in rtools. I know that those tools worked for the wheels so I used them here.

I used the 8.1.0 for the 32 bit install because the OpenBLAS 32-bit library was built with gcc 8.3.0 (the closest version). 8.1.0 is quite an old gcc version, so it made less sense to try and get the 64-bit build to work with 8.1.0.

I had to do lots of debugging in CI of the azure builds because they're quite old setups, so once I got something working I left it alone.

Reply:
`clang-cl.exe` is used by Pythran, see for example https://dev.azure.com/scipy-org/SciPy/_build/results?buildId=23257&view=logs&j=d30ccee0-62b2-572b-86e1-0fd5d4704592&t=dd6887a0-9c47-504f-8c01-87a519f6f92e. You linked to the job that disabled Pythran.

> I had to do lots of debugging in CI of the azure builds because they're quite old setups, so once I got something working I left it alone.

Okay, that sounds fine for the 32-bit builds.

Reply:
> clang-cl.exe is used by Pythran

So why doesn't Pythran complain if clang-cl isn't installed? Can it not use the C compiler from the rest of the toolchain? Is the solution here just to install clang-cl again?

Reply:
> So why doesn't Pythran complain if clang-cl isn't installed? Can it not use the C compiler from the rest of the toolchain? Is the solution here just to install clang-cl again?

It can use Mingw just fine, we should only make sure that it does not use MSVC because that leads to hard-to-debug problems.

Reply:
`setup.py == PITA`. I tried a multitude of ways when creating this PR to avoid MSVC and force the use of gcc. `setup.py` seems to ignore the environment variables `CC` and `CXX` on Windows. Anyway abe9627 seems to be using clang-cl for Pythran compilation.

Reply:
Meson is kind of the opposite - it's good with `CC` & co, but you have to work a bit to make it use MSVC because it's not on the PATH: https://dvdhrm.github.io/2021/04/21/meson-msvc-github-actions/.